internal/services/aadb2c/aadb2c_directory_resource.go

```diff
@@ -168,6 +168,10 @@ func (r AadB2cDirectoryResource) Create() sdk.ResourceFunc {
 				return fmt.Errorf("checking availability of `domain_name`: %v", err)
 			}
 
+			if availabilityResult.Model == nil {
+				return fmt.Errorf("checking availability of `domain_name`: the response from the API was nil")
+			}
+
```

Comment:
I don't think this is what we should be doing? instead we should be wrapping the code that uses this property in an if with a != nil ie
```suggestion
	if availabilityResult.Model != nil {
	// code does things
	}
```
?

Reply:
The origin logic returns an error if domain availablitity check failed. if availabilityResult.Model is nil, shouldn't we also think it as not avaliable and return an error?

Reply:
@wuxu92 the `go-azure-sdk` already handles cases where the Model is nil and will return an error - we should still nil check it in the provider to prevent crashes.

Reply:
updated this to move the login into `if a != nil` check